mod-calendar/src/main/resources/domain/mod-calendar/features/calendarPeriods.feature

```diff
@@ -4,112 +4,384 @@ Feature: Calendar periods
+    And match $.openingPeriods[0].openingDay.open == true
+    And match $.openingPeriods[0].openingDay.allDay == false
+    And match $.openingPeriods[1].openingDay.openingHour[0].startTime == '00:00'
+    And match $.openingPeriods[1].openingDay.openingHour[0].endTime == '23:59'
+    And match $.openingPeriods[1].openingDay.open == false
+    And match $.openingPeriods[1].openingDay.allDay == true
 
-  @Undefined
   Scenario: GET all periods including closed hours with 200 on success
-    * print 'undefined'
+    * def periodId = call uuid1
+    * def servicePointId = call uuid1
+    * def periodName = 'Test period'
+    * def createPeriodRequest = read('samples/createPeriod.json')
+
+    Given path 'calendar/periods/' + servicePointId + '/period'
+    And request createPeriodRequest
+    When method POST
+    Then status 201
+
+    Given path 'calendar', 'periods'
+    When method GET
+    Then status 200
+    And match $.openingPeriods[0].openingDay.open == true
+    And match $.openingPeriods[1].openingDay.open == false
 
-  @Undefined
   Scenario: GET all periods including exceptional hours with 200 on success
-    * print 'undefined'
+    * def servicePointId = call uuid1
+
+    Given path 'calendar/periods/' + servicePointId + '/period'
+    * def periodId = call uuid1
+    * def exceptionPeriodId = call uuid1
+    * def periodName = 'Test period'
+    * def createPeriodRequest = read('samples/createPeriod.json')
+    And request createPeriodRequest
+    When method POST
+    Then status 201
+
+    Given path 'calendar', 'periods'
+    When method GET
+    Then status 200
+    And match $.openingPeriods[4].date == '2030-08-06T00:00:00.000+00:00'
+    And match $.openingPeriods[4].openingDay.exceptional == false
+
+    Given path 'calendar/periods/' + servicePointId + '/period'
+    * def startExceptionDate = '2030-08-06'
+    * def endExceptionDate = '2030-08-06'
+    * def createExceptionRequest = read('samples/createException.json')
+    And request createExceptionRequest
+    When method POST
+    Then status 201
+
+    Given path 'calendar', 'periods'
+    When method GET
+    Then status 200
+    And match $.openingPeriods[4].date == '2030-08-06T00:00:00.000+00:00'
+    And match $.openingPeriods[4].openingDay.exceptional == true
 
-  @Undefined
   Scenario: GET library hours period for service point with 200 on success
-    * print 'undefined'
+    * def servicePointId = call uuid1
+
+    Given path 'calendar/periods/' + servicePointId + '/period'
+    When method GET
+    Then status 200
 
-  @Undefined
   Scenario: GET library hours period for service point with opening days and 200 on success
-    * print 'undefined'
+    * def periodId = call uuid1
+    * def servicePointId = call uuid1
+    * def periodName = 'Test period'
+    * def createPeriodRequest = read('samples/createPeriod.json')
 
-  @Undefined
-  Scenario: GET library hours period for service point including past openings and with 200 on success
```

Comment:
Revert this, add showPast query parameter

Reply:
done